sdk-base/src/main/java/com/mapbox/maps/extension/observable/ObservableExtension.kt

```diff
@@ -329,34 +329,41 @@ fun Event.getSourceDataLoadedEventData(): SourceDataLoadedEventData {
   return Gson().fromJson(json, SourceDataLoadedEventData::class.java)
 }
 
-internal fun Event.getIDStringEventData(): IDStringEventData {
-  val json = data.toJson()
-  return Gson().fromJson(json, IDStringEventData::class.java)
-}
-
 /**
  * Get the parsed event data for style missing event.
- * @return a parsed IDStringEventData.
+ * @return a parsed StyleImageMissingEventData.
  */
-fun Event.getStyleImageMissingEventData(): IDStringEventData = getIDStringEventData()
+fun Event.getStyleImageMissingEventData(): StyleImageMissingEventData {
+  val json = data.toJson()
+  return Gson().fromJson(json, StyleImageMissingEventData::class.java)
+}
 
 /**
  * Get the parsed event data for style image unused event.
- * @return a parsed IDStringEventData.
+ * @return a parsed StyleImageUnusedEventData.
  */
-fun Event.getStyleImageUnusedEventData(): IDStringEventData = getIDStringEventData()
+fun Event.getStyleImageUnusedEventData(): StyleImageUnusedEventData {
```

Comment:
we could keep `IDStringEventData` class that was defined earlier, it has the same purpose as `StyleImageUnusedEventData.kt`, `StyleImageUnusedEventData.kt`, `StyleImageMissingEventData.kt`, wdty?

Reply:
we need to make them extensible for each event, so let's separate them.